### PR TITLE
Added colorful logging

### PR DESCRIPTION
**Pull request by flying-sheep**
This will add the common dependency crayon.

shouldn’t be a problem, but maybe we should wait with new dependencies until having it on CRAN makes this trivial (i.e. we won’t have to mention dependencies in readmes anymore)

### CODE REVIEW

File: R/logging.r [hunk 1]
```diff
@@ -10,26 +10,29 @@
 #'
 #' @name log_*
 #' @param ...  message to log
+#' @importFrom crayon green
 #' @export
 log_debug <- function(...) {
     if (isTRUE(getOption('jupyter.log_level') >= 3L)) {
-        log_msg('DEBUG', sprintf(...))
+        log_msg(green('DEBUG'), sprintf(...))
```
Comment (janschulz) [inline R/logging.r]: IMO the coloring should be done in log_msg_stderror as as far as I can set will now output escape chars into the text based logfile.

**Comment by janschulz** [comment]
I'm not sure this is helpfull: the R kernel log is currently mixed together  with the normal jupyter notebook log (at least on my windows cmd window which opens when the notebook is opened). But that log isn't colored so it becomes a mix of colored and uncolored log entries.

File: R/logging.r (hunk 1 above)
Comment (flying-sheep) [inline R/logging.r]: makes sense. i’ll simply add a parameter to that function

**Comment by flying-sheep** [comment]
still helpful IMHO: better help visually distinguish a part of the logspam than nothing.

File: R/logging.r [hunk 2]
```diff
@@ -59,14 +62,15 @@ log_msg_stderror <- function(lvl, msg){
...
 log_msg_logfile <- function(lvl, msg) {
     cur_logfile <- getOption('jupyter.logfile')
     if (!is.na(cur_logfile)) {
         if (.is_changed_logfile(cur_logfile)) {
-            log_msg_stderror('INFO', sprintf('Logging to %s', cur_logfile))
+            log_msg_stderror(blue('INFO'), sprintf('Logging to %s', cur_logfile))
         }
         log_con <- file(cur_logfile, open = 'ab')
-        writeBin(charToRaw(sprintf('%s %s: %s\n', format(Sys.time()), lvl, msg)), log_con, endian = 'little')
+        writeBin(charToRaw(sprintf('%s %s: %s\n', format(Sys.time()), strip_style(lvl), msg)), log_con, endian = 'little')
```
Comment (flying-sheep) [inline R/logging.r]: i remove the styling now here

**Pull request merged**